reference/src/layout/arrays.md

```diff
@@ -0,0 +1,33 @@
+# Layout of Rust array types
+
+Array types, `[T; N]`, store `N` values of type `T` contiguously with a constant
+_stride_ (the distance between each two consecutive values).
+
+The offset of the first array element is `0`.
+
+The stride of the array is computed as the size of the element type rounded up
+to the next multiple of the alignment of the element type. That is, the _stride_
+of an array can be larger than the element size iff the alignment requirements
+of the element are larger than its size.
+
+Having a stride larger than the element size allows:
+
+```rust,ignore
+struct A(u16, u8); // size_of == 3, align_of == 4
```

Comment:
This is wrong, `size_of` is 4 here. In some sense the size *could be* 3, but it isn't. If we ever introduce a stride concept so that we can make that disctinction, backwards compatibility might force us to create a new term for the quantity that is 3 then (code might manually compute the size of an array as `size_of::<ElemTy>() * LENGTH`).

Reply:
The intro to this code says "Having a stride larger than the element size allows" -- so indeed size is not 3 today, but it could be.

Reply:
> This is wrong, size_of is 4 here.

Do we guarantee that in the layout of structs ?

Reply:
We don't guarantee the size of this struct, but the fact is, `size_of::<A>()` is not 3 on any existing Rust implementation. Saying it *is* 3 is simply wrong. If you really want to make the point that its size *could* be 3, then say *that*. As is, this section is plainly misleading.

Reply:
I did not say that it is 3, I said that for the rules to allow it to be 3 we would need to support a `stride > size`. Anyways, the way I wrote that was super-confusing, I hope it is clearer now.